src/main/java/org/betonquest/betonquest/quest/variable/point/AbstractPointVariableFactory.java

```diff
@@ -0,0 +1,102 @@
+package org.betonquest.betonquest.quest.variable.point;
+
+import org.apache.commons.lang3.tuple.Triple;
+import org.betonquest.betonquest.Instruction;
+import org.betonquest.betonquest.api.logger.BetonQuestLogger;
+import org.betonquest.betonquest.exceptions.ObjectNotFoundException;
+import org.betonquest.betonquest.exceptions.QuestException;
+import org.betonquest.betonquest.id.ID;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Locale;
+
+/**
+ * A factory for creating Point variables.
+ *
+ * @param <T> the data holder type
+ */
+@SuppressWarnings("PMD.AbstractClassWithoutAbstractMethod")
+public abstract class AbstractPointVariableFactory<T> {
+
+    /**
+     * The data holder.
+     */
+    protected final T dataHolder;
+
+    /**
+     * The logger instance for this factory.
+     */
+    protected final BetonQuestLogger logger;
+
+    /**
+     * Create a new Point variable factory.
+     *
+     * @param dataHolder the data holder
+     * @param logger     the logger instance for this factory
+     */
+    public AbstractPointVariableFactory(final T dataHolder, final BetonQuestLogger logger) {
+        this.dataHolder = dataHolder;
+        this.logger = logger;
+    }
+
+    /**
+     * Parse the instruction to get the category, type and amount of the point.
+     *
+     * @param instruction the instruction to parse
+     * @return a triple containing the category, amount and type of the point
+     * @throws QuestException if the instruction could not be parsed
+     */
+    protected Triple<String, Integer, PointCalculationType> parseInstruction(final Instruction instruction) throws QuestException {
+        final String category = getCategory(instruction);
+        final PointCalculationType type = getType(instruction);
+        int amount = 0;
+        if (type == PointCalculationType.LEFT) {
+            amount = getAmount(instruction);
+        }
+        return Triple.of(category, amount, type);
+    }
+
+    @SuppressWarnings({"PMD.AvoidLiteralsInIfCondition", "PMD.ShortVariable"})
```

Comment:
I think We dont want to suppress ShortVariable, you should think about an better name like packageId